tests/unit/hive_metastore/test_locations.py

```diff
@@ -226,3 +230,308 @@ def test_match_table_external_locations():
     assert len(matching_locations) == 1
     assert ExternalLocation("gcs://test_location2/a/b/", 1) in missing_locations
     assert ExternalLocation("abfss://cont1@storagetest1/a/", 2) in missing_locations
+
+
+def test_mount_listing_multiple_folders():
+    client = create_autospec(WorkspaceClient)
+
+    first_folder = FileInfo("dbfs:/mnt/test_mount/table1/", "table1/", "", "")
+    second_folder = FileInfo("dbfs:/mnt/test_mount/table2/", "table2/", "", "")
+    folder_table1 = FileInfo("dbfs:/mnt/test_mount/table1/_delta_log/", "_delta_log/", "", "")
+    folder_table2 = FileInfo("dbfs:/mnt/test_mount/table2/_SUCCESS", "_SUCCESS", "", "")
+    folder_table3 = FileInfo("dbfs:/mnt/test_mount/table2/1.snappy.parquet", "1.snappy.parquet", "", "")
+
+    def my_side_effect(path, **_):
+        if path == "/mnt/test_mount":
+            return [first_folder, second_folder]
+        if path == "dbfs:/mnt/test_mount/table1/":
+            return [folder_table1]
+        if path == "dbfs:/mnt/test_mount/table2/":
+            return [folder_table2, folder_table3]
+        return None
+
+    client.dbutils.fs.ls.side_effect = my_side_effect
+    backend = MockBackend(
+        rows={
+            'hive_metastore.test.tables': [],
+            'test.mounts': MOUNT_STORAGE[("/mnt/test_mount", "adls://bucket/")],
+        }
+    )
+    mounts = Mounts(backend, client, "test")
+    results = TablesInMounts(backend, client, "test", mounts).snapshot()
+    assert results == [
+        Table("hive_metastore", "mounted_/mnt/test_mount", "table1", "EXTERNAL", "DELTA", "adls://bucket/table1"),
+        Table("hive_metastore", "mounted_/mnt/test_mount", "table2", "EXTERNAL", "PARQUET", "adls://bucket/table2"),
```

Comment:
```suggestion
        Table("hive_metastore", "mounted_test_mount", "table2", "EXTERNAL", "PARQUET", "adls://bucket/table2"),
```
No slashes in schema name!